substra/sdk/client.py

```diff
@@ -311,6 +311,15 @@ def __init__(
             )
             self._token = self.login(config_dict["username"].value, config_dict["password"].value)
 
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        self.logout()
+
+    def __del__(self):
+        self.logout()
```

Comment:
Unsure whether context manager is still needed with `__del__` defined, but people online say not to rely on it and using `with` is preferable. However that does mean updating our documentation.

Reply:
What part of the documentation are you reffering to ?

Reply:
I mean that in the documentation all examples of the form:

```
client = Client(...)
do_stuff(client)
```

should be replaced with:


```python
with Client(...) as client:
    do_stuff(client)
```

Reply:
Some scripts (such as the example) need the client for more than a thousand lines, no I would not replace the old version, maybe just indicate that the context manager is also a valid option ?

Reply:
I added it in the `sdk.Client` reference, under the `username` argument.